api/v1alpha1/common_types.go

```diff
@@ -91,6 +94,18 @@ type ForgejoRepo struct {
 	Name string `json:"name"`
 }
 
+// BitbucketCloudRepo is a repository in Bitbucket Cloud, identified by its workspace and repository.
+type BitbucketCloudRepo struct {
+	// Workspace is the workspace of the repository.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Workspace string `json:"workspace"`
+	// Repository is the repository of the repository.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
```

Comment:
Unlike owner, turns out the max length of this is actually well documented: https://support.atlassian.com/bitbucket-cloud/docs/create-a-repository/

Based on some light testing, I think the given regex should cover the possible characters. We can adjust later if it's too restrictive.

```suggestion
	// +kubebuilder:validation:MaxLength=62
	// +kubebuilder:validation:Pattern="^[a-zA-Z0-9_-]+.$"
```